internal/common/connectionsapi/client_test.go

```diff
@@ -0,0 +1,290 @@
+package connectionsapi
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_sets_auth_token_and_content_type(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "Bearer some token", r.Header.Get("Authorization"))
+		assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
+		_, _ = fmt.Fprintf(w, `{}`)
+	}))
+	defer svr.Close()
+
+	c, err := NewClient("some token", svr.URL, nil)
+	require.NoError(t, err)
+	_, err = c.CreateMetricsEndpointScrapeJob(context.Background(), "some stack id", MetricsEndpointScrapeJob{})
+	require.NoError(t, err)
+}
+
+func TestClient_CreateMetricsEndpointScrapeJob_sends_request_and_receives_response(t *testing.T) {
```

Comment:
nit: Can we organise these tests into parent / child subtests so that it's easy to run all the tests for a given func in one pass?

e.g.
```
func TestClient_CreateMetricsEndpointScrapeJob(t *testing.T) {
  t.Run("successfully creates job", func...)
  t.Run("returns error given 500 from API", func...)
}
```